pmd-cpp/src/test/java/net/sourceforge/pmd/cpd/CPPTokenizerTest.java

```diff
@@ -159,6 +159,31 @@ public void testLexicalErrorFilename() throws Exception {
         tokenizer.tokenize(code, new Tokens());
     }
 
+    @Test
+    public void testStringPrefixes() {
+        final String code = "char     a =  '\\x30';         // character, no semantics" + PMD.EOL
+                + "wchar_t  b = L'\\xFFEF';       // wide character, no semantics" + PMD.EOL
+                + "char16_t c = u'\\u00F6';       // 16-bit, assumed UTF16?" + PMD.EOL
+                + "char32_t d = U'\\U0010FFFF';   // 32-bit, assumed UCS-4" + PMD.EOL
+                + "char     A[] =  \"Hello\\x0A\";         // byte string, \"narrow encoding\"" + PMD.EOL
+                + "wchar_t  B[] = L\"Hell\\xF6\\x0A\";      // wide string, impl-def'd encoding" + PMD.EOL
+                + "char16_t C[] = u\"Hell\\u00F6\";        // (1)" + PMD.EOL
+                + "char32_t D[] = U\"Hell\\U000000F6\\U0010FFFF\"; // (2)" + PMD.EOL
+                + "auto     E[] = u8\"\\u00F6\\U0010FFFF\"; // (3)";
+        Tokens tokens = parse(code);
+        assertTrue(TokenEntry.getEOF() != tokens.getTokens().get(0));
+        assertEquals(58, tokens.size());
```

Comment:
Maybe we should make the code sample smaller and verify the actual token types? Otherwise, we won't know, whether the raw string literal is actually tokenized correctly. Same with the characters...

Reply:
I'm not sure how to check the type of the token in a unit test. Maybe it's sufficient to verify that the whole character or string literal belongs to a single token.

Reply:
Yes, that's what I did - using tokenEntry.toString() to see, what the original matched image was and use that to verify it.